core/chains/evm/gas/universal_estimator.go

```diff
@@ -0,0 +1,412 @@
+package gas
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	bigmath "github.com/smartcontractkit/chainlink-common/pkg/utils/big_math"
+
+	commonfee "github.com/smartcontractkit/chainlink/v2/common/fee"
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/assets"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/gas/rollups"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+// metrics are thread safe
+var (
+	promUniversalEstimatorGasPrice = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "gas_price_updater",
+		Help: "Sets latest gas price (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+	promUniversalEstimatorBaseFee = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "base_fee_updater",
+		Help: "Sets latest BaseFee (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+	promUniversalEstimatorMaxPriorityFeePerGas = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "max_priority_fee_per_gas_updater",
+		Help: "Sets latest MaxPriorityFeePerGas (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+	promUniversalEstimatorMaxFeePerGas = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "max_fee_per_gas_updater",
+		Help: "Sets latest MaxFeePerGas (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+)
+
+const (
+	MinimumBumpPercentage   = 10 // based on geth's spec
+	ConnectivityPercentile  = 80
+	BaseFeeBufferPercentage = 40
```

Comment:
IIUC this is added on top of the current base fee to account for increases to the base fee between our estimation and when the tx is broadcasted/included. Is there any reason this isn't a more exact value since we know base fee is only adjusted by 12.5% every block? Hope I'm not making it too complicated when the point is to simplify but wouldn't it be better to instead set a `BaseFeeBufferBlocks=3` then we use a formula like `baseFee * (1.125 ^ BaseFeeBufferBlocks)`?

Reply:
This was intentional. `EIP1559FeeCapBufferBlocks` gets its value from BumpThreshold + 1 -> 4, but it has never been changed in our configs AFAIK. Adding another config without knowing whether someone is going to use it adds complexity and keep in mind someone has to configure it and be aware of how it works.
Regarding the value, I wanted to set it to something that works but doesn't follow the calculation you posted so we don't go into the discussion of making it configurable, but I guess that ship has sailed haha. Also, 12.5% is only for Ethereum, some chains have different values but I didn't want to introduce that complexity if I don't have to. Perhaps we can revisit this value during testing.

Reply:
Agree we probably shouldn't make that a new config. I was thinking it'd just be a fixed value that replaces `BaseFeeBufferPercentage`. But I'm ok with just leaving this since this works based on your soak tests. I think FeeHistory already provides a base fee for the next block so regardless we'll always be a few blocks ahead with our price even if it's this simple.